spec/controllers/events_controller_spec.rb

```diff
@@ -1,29 +1,177 @@
+      expect(event.summary).to eq('Summary about event')
+      expect(event.targets.count).to eq(2)
+
+      expect(response_json['status']).to eq('created')
     end
 
-    it 'send email when create event' do
-      params = { event: {
-        description: 'some description',
-        start_date: Time.current - 1.day,
-        finish_date: Time.current, user_id: admin.id
-      } }
+    it 'send email when created event' do
+      expect do
+        post :create, params: { event: event_params }, format: :json
+      end.to change { ActionMailer::Base.deliveries.count }.by(1)
+    end
 
-      expect { post :create, params: params, format: :json }.to change { ActionMailer::Base.deliveries.count }.by(1)
-      expect(Event.where(description: 'some description')).to exist
+    it 'return error when finish date less than start date' do
+      expect do
+        post(
+          :create,
+          params: { event: event_params.merge(finish_date: event_params[:start_date] - 2.days) },
+          format: :json
+        )
+      end.to change(Event, :count).by(0)
+      expect(response_json['finish_date']).to include("can't be less than start date")
     end
 
     it 'do not send email when failed create event' do
-      params = { event: { description: 'some description', finish_date: Time.current } }
+      expect do
+        post(
+          :create,
+          params: { event: event_params.merge(finish_date: nil) },
+          format: :json
+        )
+      end.to change { ActionMailer::Base.deliveries.count }.by(0)
+    end
+  end
+
+  describe 'PUT :update' do
+    let(:event) { create(:event, targets: create_list(:target, 2)) }
+
+    let(:targets_params) do
+      [
+        {
+          id: event.targets.first.id,
+          description: 'Test data 1'
+        },
+        {
+          id: event.targets.last.id,
+          description: 'Test data 2'
+        }
+      ]
+    end
+
+    before { user.events << event }
 
-      expect { post :create, params: params, format: :json }.to change { ActionMailer::Base.deliveries.count }.by(0)
-      expect(Event.where(description: 'some description')).to_not exist
+    it 'update with valid params' do
+      expect do
+        put(
+          :update,
+          params: { event: { description: 'New', targets_attributes: targets_params }, id: event.id },
+          format: :json
+        )
+      end.to change(Event, :count).by(0)
+
+      event.reload
+      expect(user.events.last.description).to eq('New')
+      expect(user.events.last.status).to eq(event.status)
+      expect(user.events.last.start_date).to eq(event.start_date)
+      expect(user.events.last.finish_date).to eq(event.finish_date)
+      expect(user.events.last.comments).to eq(event.comments)
+      expect(user.events.last.summary).to eq(event.summary)
+      expect(user.events.last.targets.count).to eq(2)
+      expect(user.events.last.targets.first.description).to eq('Test data 1')
```

Comment:
`expect(event.targets.map(&:description)).to include('Test data 1', 'Test data 2')`

Reply:
Do not use `first` and `second`, because if change order, test will be failed.